Source/Extensions/Blazorise.LoadingIndicator/LoadingIndicatorService.cs

```diff
@@ -0,0 +1,196 @@
+                if ( indicators.Count == 0 )
+                {
+                    SingleMode( indicator );
+                }
+                else if ( indicators.Count == 1 )
+                {
+                    MultiMode();
+                }
+
+                indicators.Add( indicator );
+            }
+        }
+
+        /// <summary>
+        /// Unsubscribe indicator from change events and remove reference
+        /// </summary>
+        /// <param name="indicator"></param>
+        void ILoadingIndicatorService.Unsubscribe( LoadingIndicator indicator )
+        {
+            lock ( hashLock )
+            {
+                indicators.Remove( indicator );
+
+                if ( indicators.Count == 0 )
+                {
+                    MultiMode();
+                }
+                else if ( indicators.Count == 1 )
+                {
+                    SingleMode( indicators.First() );
+                }
+            }
+        }
+
+        private void SetBusyMulti( bool value )
+        {
+            lock ( hashLock )
+            {
+                foreach ( var indicator in indicators )
+                {
+                    indicator.SetBusy( value );
+                }
+            }
+        }
+
+        private void SetLoadedMulti( bool value )
+        {
+            lock ( hashLock )
+            {
+                foreach ( var indicator in indicators )
+                {
+                    indicator.SetLoaded( value );
+                }
+            }
+        }
+
+        private bool? GetBusyMulti()
+        {
+            bool? val = null;
+            lock ( hashLock )
+            {
+                foreach ( var indicator in indicators )
+                {
+                    if ( val == null )
+                    {
+                        val = indicator.Busy;
+                    }
+                    else
+                    {
+                        if ( val != indicator.Busy )
+                        {
+                            return null;
+                        }
+                    }
+                }
+            }
+            return val;
+        }
+
+        private bool? GetLoadedMulti()
+        {
+            bool? val = null;
+            lock ( hashLock )
+            {
+                foreach ( var indicator in indicators )
+                {
+                    if ( val == null )
+                    {
+                        val = indicator.Loaded;
+                    }
+                    else
+                    {
+                        if ( val != indicator.Loaded )
+                        {
+                            return null;
+                        }
+                    }
+                }
+            }
+            return val;
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Returns Busy state shared by all indicator instances or null if the state is not the same for all indicators.
```

Comment:
Public comments can be moved to the interface and in implementation we just add `/// <inheritdoc/>`

Reply:
Also, having a bool meaning multiple things seems kinda strange. Maybe a better approach is to have an enum?

eg

```
LoadingBusyState
{
  All,
  Partial,
  None,
}
```

Reply:
BTW, is `Busy` name still valid? Since we now have Show/Hide methods should we rename it to `Visible`?

Reply:
> should we rename it to `Visible`

Yes, I was thinking about that too, I'll change it. Thanks! 


> Maybe a better approach is to have an enum?

I think it's pretty standard to use bool? for yes/no/maybe for example WPF checkbox/toggle https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.primitives.togglebutton.ischecked?view=windowsdesktop-6.0

Reply:
> I think it's pretty standard to use bool? for yes/no/maybe for example WPF checkbox/toggle

OK, maybe we can leave it `bool?` then.

Reply:
for Visible, looking a bit closer, looks like there's already a Visibility property, but it's not a boolean. Maybe use "bool Loading" instead of visible? Visible woudl match show/hide better, but since there's already Visibility, it may be confusing.. Any other good names? or can leave it at Busy.. Thoughts?

Reply:
We already have Show/Hide/Visible on other components like Modal, Dropdown, etc., so it would not be problematic. But, Start/Stop/Active is also a good alternative.